tests/unittest_manager.py

```diff
@@ -325,6 +325,24 @@ def hook(modname: str):
             with self.assertRaises(AstroidBuildingError):
                 self.manager.ast_from_module_name("foo.bar.baz")
 
+    def test_same_name_import_module(self) -> None:
+        """test inference of an import statement with the same name as the module.
+
+        see https://github.com/pycqa/pylint/issues/7289.
+        """
+        data_dir = resources.find_in_testdata("import_conflicting_names")
+        math_file = data_dir / "math.py"
```

Comment:
```suggestion
        math_file = resources.construct_absolute_path_in_testdata("import_conflicting_names/math.py")
```
I think this would work ?